.github/actions/test-wasi/action.yml

```diff
@@ -0,0 +1,43 @@
+name: Test
+inputs:
+  testArtifacts:
+    default: null
+    required: false
+  runTestsParameters:
+    default: ''
+    required: false
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      run: |
+        cat <<-'EOF' > /opt/bin/run-tests.sh
+        	#!/bin/sh
+        	WASMTIME_BACKTRACE_DETAILS=1 /opt/bin/wasmtime run --mapdir /::/ $PHP_ROOT_PATH/sapi/cli/php.optimized 2> /dev/null -- "$@"
+        EOF
+        chmod +x /opt/bin/run-tests.sh
+    - shell: bash
+      run: |
+        set -x
+        export PHP_ROOT_PATH=$PWD
+        export SKIP_IO_CAPTURE_TESTS=1
+        export TEST_PHP_JUNIT=junit.out.xml
+        export STACK_LIMIT_DEFAULTS_CHECK=1
+        php run-tests.php -q ${{ inputs.runTestsParameters }} \
+          -p /opt/bin/run-tests.sh \
+          -j$(/usr/bin/nproc) \
+          -g FAIL,BORK,LEAK,XLEAK \
+          --no-progress \
+          --offline \
+          --show-diff \
+          --show-slow 1000 \
+          --set-timeout 120 \
+          tests/lang \
```

Comment:
whole suite incl. extensions should be run, incompatible tests should be skipped individually